src/main/java/org/opensearch/flowframework/workflow/ReindexStep.java

```diff
@@ -0,0 +1,175 @@
+            }
+            if (requestsPerSecond != null) {
+                reindexRequest.setRequestsPerSecond(requestsPerSecond);
+            }
+            if (requireAlias != null) {
+                reindexRequest.setRequireAlias(requireAlias);
+            }
+            if (maxDocs != null) {
+                reindexRequest.setMaxDocs(maxDocs);
+            }
+            if (slices != null) {
+                reindexRequest.setSlices(slices);
+            }
+
+            ActionListener<BulkByScrollResponse> actionListener = new ActionListener<>() {
+
+                @Override
+                public void onResponse(BulkByScrollResponse bulkByScrollResponse) {
+                    logger.info("Reindex from source: {} to destination {}", sourceIndices, destinationIndex);
+                    try {
+                        if (bulkByScrollResponse.getBulkFailures().isEmpty() && bulkByScrollResponse.getSearchFailures().isEmpty()) {
+                            reIndexFuture.onResponse(
+                                new WorkflowData(
+                                    Map.of(
+                                        NAME,
+                                        Map.ofEntries(
+                                            Map.entry(DESTINATION_INDEX, destinationIndex),
+                                            Map.entry(SOURCE_INDICES, sourceIndices)
+                                        )
+                                    ),
+                                    currentNodeInputs.getWorkflowId(),
+                                    currentNodeInputs.getNodeId()
+                                )
+                            );
+                        } else {
+                            String errorMessage = "Failed to get bulk response " + bulkByScrollResponse.getBulkFailures();
+                            reIndexFuture.onFailure(new FlowFrameworkException(errorMessage, RestStatus.BAD_REQUEST));
+                        }
+                    } catch (Exception e) {
+                        String errorMessage = "Failed to parse and update new created resource";
+                        logger.error(errorMessage, e);
+                        reIndexFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(e)));
+                    }
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    String errorMessage = "Failed to reindex from source " + sourceIndices + " to " + destinationIndex;
+                    logger.error(errorMessage, e);
+                    reIndexFuture.onFailure(new WorkflowStepException(errorMessage, ExceptionsHelper.status(e)));
+                }
+            };
+
+            client.execute(ReindexAction.INSTANCE, reindexRequest, actionListener);
+
+        } catch (IllegalArgumentException iae) {
+            reIndexFuture.onFailure(new WorkflowStepException(iae.getMessage(), RestStatus.BAD_REQUEST));
```

Comment:
I like this, possibly add some text before the parsing exception message identifying which step or step id this failure is occurring in?